.github/workflows/image.yml

```diff
@@ -0,0 +1,59 @@
+## Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.## 
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0 
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## 
+
+name: image
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'release-*'
+  push:
+    tags:
+      - 'v*.*.*'  
+    branches:
+      - 'main'
+      - 'release-*'
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/nvidia/holodeck
+          tags: |
+            type=semver,pattern={{raw}}
+            type=ref,event=branch
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Do we only want to trigger this on tags, or for shas as well? As a matter of interest, what is the version set to if this is not a tag?

Reply:
as is  image will be push only on tags and push to main. not during pull request, since I am setting the actions.yaml to point to main for now. moving forward I will point to a tagged version, but since this action is new and under development main will be moving at least for this and next week until we can call a stable tag